### PR TITLE
Add basic CRUD support for Lambda VpcConfig

### PR DESCRIPTION
**Pull request by dominikschubert**
Addresses #8556

### CODE REVIEW

File: tests/integration/awslambda/test_lambda_api.py [hunk 1]
```diff
@@ -573,6 +574,110 @@ def test_list_functions(self, create_lambda_function, lambda_su_role, snapshot,
...
+        snapshot.add_transformer(
+            snapshot.transform.regex(security_group_id_1, "<security_group_id_1>")
+        )
+        snapshot.add_transformer(
+            snapshot.transform.regex(security_group_id_2, "<security_group_id_2>")
+        )
+
+        cleanups.append(
+            lambda: aws_client.awslambda.delete_function(FunctionName=function_name)
+        )  # needed because otherwise VPC is still linked to function and deletion is blocked
```
Comment (dominikschubert) [inline tests/integration/awslambda/test_lambda_api.py:622]: Unfortunately this still doesn't work properly on AWS and the last subnet/security group + vpc are leaking. Not critical for this PR but we probably need to wait for the lambda to actually  disappear before this is detached an can be deleted :thinking:

File: tests/integration/templates/cfn_parameter_list_type.yaml [hunk 2]
```diff
@@ -1,29 +1,14 @@
...
     Type: List<String>
-  SecurityGroupId:
-    Type: String
-  FunctionRole:
-    Type: String
 Resources:
-  Function76856677:
-    Type: AWS::Lambda::Function
+  MyParam:
+    Type: AWS::SSM::Parameter
```
Comment (joe4dev) [inline tests/integration/templates/cfn_parameter_list_type.yaml:8]: nice test simplification 👍

File: tests/integration/awslambda/test_lambda_api.py (hunk 1 above)
Comment (joe4dev) [reply tests/integration/awslambda/test_lambda_api.py:622]: VPC testing is annoying 🐌 

nit: would be good to add the comment in the test such that it does not get lost in this PR discussion
💡 slow tests + improper cleanup are candidates for test labels

**Review by joe4dev — APPROVED** [review]
LGTM 👍 

Lambda model updates break CloudPods. Good to get this PR in now with a couple of recent Lambda model updates.

**Pull request merged**